src/hdf5_dl.c

```diff
@@ -449,3 +468,7 @@ herr_t H5Zregister(const void *cls)
 {
 CALL(-1, H5Zregister, cls)
 }
+
+herr_t H5Zunregister(H5Z_filter_t id){
+    CALL(0, H5Zunregister, id)
```

Comment:
Use a negative value for fallback as it is what this function returns in case of error:

```suggestion
    CALL(-1, H5Zunregister, id)
```